pyanaconda/modules/users/users.py

```diff
@@ -217,8 +220,14 @@ def _ksdata_to_user_data(self, user_ksdata):
         # We need to make sure we correctly convert between these two.
         if user_ksdata.uid is not None:
             user_data.uid = user_ksdata.uid
+            user_data.uid_status = ""
+        else:
+            user_data.uid_status = ID_NOT_SET
```

Comment:
You can simplify this by abstracting this logic into the structure class. It would still have data as strings but we don't have to bother when working with those.

Reply:
Ideally, there would be methods `set_uid` and `get_uid` that would allow to set and get `None`. It will require a lot of additional code, because we need the same set of methods three times, but I don't know about a better solution.

Reply:
Good idea.

Reply:
Done (I think).